core/templates/components/forms/schema-based-editors/schema-based-dict-editor.component.spec.ts

```diff
@@ -0,0 +1,149 @@
+import { IdGenerationService } from 'services/id-generation.service';
+import { SchemaBasedDictEditorComponent } from './schema-based-dict-editor.component';
+
+describe('Schema Based Dict Editor Component', () => {
+  let component: SchemaBasedDictEditorComponent;
+  let fixture: ComponentFixture<SchemaBasedDictEditorComponent>;
+  let idGenerationService: IdGenerationService;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [FormsModule],
+      declarations: [
+        SchemaBasedDictEditorComponent
+      ],
+      providers: [
+        FocusManagerService,
+        IdGenerationService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SchemaBasedDictEditorComponent);
+    component = fixture.componentInstance;
+    idGenerationService = TestBed.inject(IdGenerationService);
+  });
+
+  it('should set component properties on initialization', fakeAsync(() => {
+    let mockFunction = function(value: number) {
+      return value;
+    };
+    component.registerOnChange(mockFunction);
+    component.registerOnTouched();
+
+    expect(component).toBeDefined();
+    expect(component.validate(null)).toEqual({});
+    expect(component.onChange).toEqual(mockFunction);
+    expect(component.onChange(true)).toEqual(true);
+  }));
+
+  it('should set directive properties on initialization', () => {
+    component.propertySchemas = [
+      {
+        name: 'Name1'
+      },
+      {
+        name: 'Name2'
+      }
+    ];
+    spyOn(idGenerationService, 'generateNewId')
+      .and.returnValues('id1', 'id2');
+
+    expect(component.fieldIds).toEqual({});
+
+    component.ngOnInit();
+
+    expect(component.fieldIds).toEqual(
+      {
+        Name1: 'id1',
+        Name2: 'id2'
+      }
+    );
+  });
+
+  it('should write value', () => {
+    component.localValue = null;
+    component.writeValue(null);
+
+    expect(component.localValue).toEqual(null);
+
+    component.writeValue(true);
+    expect(component.localValue).toBeTrue();
+  });
+
+  it('should update value when local value change', () => {
+    component.localValue = {
+      first: 'true'
+    };
+
+    expect(component.localValue.first).toEqual('true');
+
+    component.updateValue('false', 'first');
+
+    expect(component.localValue.first).toEqual('false');
+  });
+
+  it('should not update value when local value not change', () => {
+    component.localValue = {
+      first: 'true'
+    };
+
+    expect(component.localValue.first).toEqual('true');
+
+    component.updateValue('true', 'first');
+
+    expect(component.localValue.first).toEqual('true');
+  });
+
+  it('should get empty object on validating', () => {
+    expect(component.validate(null)).toEqual({});
+  });
+
+  it('should get schema', () => {
+    component.propertySchemas = [
+      {
+        name: 'id1',
+        schema: 'schema'
```

Comment:
This isn't a valid schema. See https://github.com/oppia/oppia/wiki/Schemas for the relevant definitions.

Reply:
change to html schema